rero_ils/modules/holdings/api.py

```diff
@@ -732,15 +732,13 @@ class HoldingsIndexer(IlsRecordsIndexer):
 
     record_cls = Holding
 
-    def index(self, record):
-        """Indexing a holding record.
+    def after_index_record(self, record):
+        """After bulk index.
 
-        Parent document is indexed as well.
+        :param record: indexec record.
         """
-        return_value = super().index(record)
         document = Document.get_record_by_pid(record.document_pid)
         document.reindex()
```

Comment:
It would be great to have an easy way to decide if we do a sync indexing or an async indexing.

Reply:
Because this function is automatically called from the Base class, we have to find a way to pass a parameter to the index and bulk index function to do this.